### PR TITLE
Add requirements validation

### PR DESCRIPTION
**Pull request by MartinHjelmare**
- Add a CI job to validate the requirements of the custom integration manifest.
- The validation checks:
  - That the manifest can be downloaded.
  - That the manifest requirements can be installed.
  - That the requirements don't install modules that are already available in the standard library (3.8).

### CODE REVIEW

File: azure-pipelines-ci.yml [hunk 1]
```diff
@@ -4,39 +4,51 @@ trigger:
...
+    pool:
+      vmImage: "ubuntu-latest"
+    steps:
+      - task: UsePythonVersion@0
+        displayName: "Use Python 3.8"
+        inputs:
+          versionSpec: "3.8"
+      - script: python -m pip install pipdeptree stdlib-list
+        displayName: "Install script dependencies"
+      - script: git diff HEAD master --name-only | scripts/validate_requirements.py
```
Comment (MartinHjelmare) [inline azure-pipelines-ci.yml:53]: I need to find a way to get the changed files in the PR. This isn't working.

File: scripts/validate_requirements.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        if not manifest:
+            validated_ok = False
+            continue
+        print()
+        print(f"Validating {manifest['domain']}:")
+        requirements = set(manifest["requirements"])
+        requirements_ok = validate_requirements(requirements)
+        if requirements_ok:
+            print("OK!")
+        validated_ok = requirements_ok
```
Comment (MartinHjelmare) [inline scripts/validate_requirements.py:46]: This should consider the existing validation state too. If that is False the result should be False too.
```suggestion
        validated_ok = validated_ok and requirements_ok
```

File: azure-pipelines-ci.yml (hunk 1 above)
Comment (ludeeus) [reply azure-pipelines-ci.yml:53]: `git diff HEAD origin/master --name-only` probably

Comment (frenck) [reply azure-pipelines-ci.yml:53]: This isn't working, because the checkouts are shallow. There are no other branches.

Comment (MartinHjelmare) [reply azure-pipelines-ci.yml]: I switched to github actions.

**Comment by MartinHjelmare** [comment]
Now it's working:
https://github.com/home-assistant/wheels-custom-integrations/pull/131/checks?check_run_id=1047731215

**Comment by pvizeli** [comment]
Do we need check/install if the requirements can be installed? Because that would only make sense on our HA base images with include the buildtools which we use on the wheel build image

**Comment by MartinHjelmare** [comment]
We need to install all requirements to resolve them recursively.

**Review by pvizeli — APPROVED** [review]

**Pull request merged**